data/modules/Debug/DebugCommodityPrices.lua

```diff
@@ -276,37 +262,90 @@ local function show_details_on_commodity(commodities, clicked)
 	ui.text(string.upper(clicked))
 	local c = commodities[clicked]
 
-	ui.text("Current system:\t " .. Game.system.name)
-	ui.text("Baseprice:\t " .. Format.Money(c.baseprice))
-	ui.text("Median price:\t " .. Format.Money(c:median()))
-	ui.text("Average price:\t " .. Format.Money(c:mean()))
-	ui.text("Standard deviation:\t " .. string.format("%.2f", c:std()))
-	ui.text("")
-
-	show_system_info(c.min, c.minsys, "Lowest")
-	show_system_info(c.max, c.maxsys, "Highest")
-
-	local diff = c.max - c.min
-	ui.text("Absolute price difference/spread:\t " .. Format.Money(diff))
-	ui.text("Relative price difference:\t " .. string.format("%.2f", 100*diff/c.min) .. " %")
-	local dist = string.format("%.2f", c.minsys:DistanceTo(c.maxsys))
-	ui.text("Distance between mininum and maximum systems:\t " .. dist .. " ly")
-
-	local data = c:get_bins()
-	local y_max = math.max(table.unpack(data))
-	ui.dummy(Vector2(0,50))
-
-	ui.plotHistogram("##price distribution", data, #data, 0, "", 0, y_max, Vector2(400, 100))
-	ui.text("Min: " .. Format.Money(c.min))
-	ui.text("Max: " .. Format.Money(c.max))
+	-- If previous click was non-purchasalbe item followed by
+	-- filtering them out, 'clicked' value will no longer be in
+	-- 'commodities'
+	if not c then
+		return
+	end
+
+	local purchasable = Commodities[c.id_name].purchasable
+
+	ui.text("Current system:\t" .. Game.system.name)
+	ui.text("Tag:\t" .. c.id_name)
+	ui.text("Purchasable:\t" .. tostring(purchasable))
+	ui.text("Baseprice:\t" .. Format.Money(c.baseprice))
+	if not purchasable then return purchasable end
+	ui.text("Median price:\t" .. Format.Money(c:median()))
+	ui.text("Average price:\t" .. Format.Money(c:mean()))
+	ui.text("Standard deviation:\t" .. string.format("%.2f", c:std()))
+
+	if ui.collapsingHeader("Lowest", {"DefaultOpen"}) then
+		show_system_info(c.min, c.minsys, "Lowest")
+	end
+	if ui.collapsingHeader("Highest", {"DefaultOpen"}) then
+		show_system_info(c.max, c.maxsys, "Highest")
+	end
+
+	if ui.collapsingHeader("Difference", {"DefaultOpen"}) then
+		local diff = c.max - c.min
+		ui.text("Absolute price difference/spread:\t " .. Format.Money(diff))
+		ui.text("Relative price difference:\t " .. string.format("%.2f", 100*diff/c.min) .. " %")
+		local dist = string.format("%.2f", c.minsys:DistanceTo(c.maxsys))
+		ui.text("Distance between mininum and maximum systems:\t " .. dist .. " ly")
+
+		local data = c:get_bins()
+		local y_max = math.max(table.unpack(data))
+		ui.dummy(Vector2(0,50))
+
+		ui.plotHistogram("##price distribution", data, #data, 0, "", 0, y_max, Vector2(400, 100))
+		ui.text("Min: " .. Format.Money(c.min))
+		ui.text("Max: " .. Format.Money(c.max))
+	end
+	return purchasable
 end
 
 
+local function station_economy(commodities, clicked, station)
```

Comment:
@Web-eWorks If you intend to review this PR, then this function is the one where I'm half stumbling around trying to understand the economy system. 

Basically, what info is meaningful to display, are the ones I have correctly named?

For completeness, I'll add this uncommitted comment I have on my local copy, adapted from IRC conversation the other day:
```
	-- GetCommodityStockFromFlow() returns "equilibrium" stock/demand,
	-- and price of the commodity is stored as a "pricemod", a
	-- percent-of-100 offset to the commodity's base price to compute
	-- market price for e.g. oxygen (substitute any commodity ID),
	-- you'd call
	--
	--    Economy.UpdateCommodityPriceMod(sBody, "oxygen", { stock, demand, pricemod }),
	--
	-- which stores the new pricemod in the table passed as the 3rd argument.
	-- You can then call
	--
	--    Economy.GetMarketPrice(commodity.price, pricemod)
	--
	-- to get the credit cost of the commodity
```

Reply:
This looks pretty good. Some of this information could be sourced from the commodity market datastore directly (`station:GetCommodityMarket()`) rather than through `GetCommodityStock()` et. al., though the only time that matters is in the case of a custom commodity price being set by a mission module.

Another thing to look at would potentially be to inspect the persistent market cache (`stationMarket` in Economy.lua), though you'd need an accessor to retrive the variable outside of the Economy module.

If you wanted to get fancier, you could provide a button that simulates the effect of restocking ticks on the commodity market - possibly make a copy of the market datastore returned from `GetCommodityMarket()` and call:

```lua
Economy.UpdateStationCommodityMarket(sBody, Engine.rand, marketCopy, comm.id_name, ticksToSimulate)
```

You could then display some sort of graph showing the commodity stock/demand levels over the simulate time - that could be a very useful tool when doing further work on the economy simulation.

Reply:
thanks for suggestions. As my main driving motivation to work on this was when fixing  #5859, I'll "call it a day" as for features, but keep your suggestions in mind for next time I'll want to dive into commodity balance.